engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -335,13 +338,31 @@ class WorkflowManagerActor(backend: Backend, config: Config)
       workflowState <- globalDataAccess.getWorkflowState(workflowId)
       // TODO: This assertion could also be added to the db layer: "In the future I'll fail if the workflow doesn't exist"
       _ <- WorkflowMetadataBuilder.assertWorkflowExistence(workflowId, workflowState)
-      workflowDescriptor <- globalDataAccess.getWorkflow(workflowId)
-      executionStatuses <- globalDataAccess.getExecutionStatuses(workflowId)
-    } yield WorkflowMetadataBuilder.workflowStdoutStderr(workflowId, backend, workflowDescriptor, executionStatuses)
+      callLogOutputs <- globalDataAccess.getCallLogOutputs(workflowId)
+    } yield workflowStdoutStderr(callLogOutputs)
   }
 
+  private def workflowStdoutStderr(callLogOutputs: Traversable[SymbolStoreEntry]): WorkflowLogs = {
+    val callsToPaths = for {
+      (key, entries) <- callLogOutputs.groupBy(ExecutionDatabaseKey.toCallLogKey)
+    } yield key -> key.toCallLogs(entries)
+
+    /* Some FP "magic" to transform the pairs of (key, logs) into the final result:
+    grouped by FQNS, ordered by shards, and then ordered by attempts */
+    callsToPaths.toList filter {
+      _._2.isDefined
+    } groupBy {
+      _._1.fqn
+    } mapValues {
+      key => key.groupBy(_._1.index).values.toIndexedSeq.sortBy(_.head._1.index) map {
+        _.sortBy(_._1.attempt).map(_._2.get).toIndexedSeq
+      }
+    }
+  }
```

Comment:
I see below and remember now this was preexisting so don't feel obligated :smile: